src/legacy/ui/public/vis/map/service_settings.js

```diff
@@ -154,31 +157,57 @@ uiModules.get('kibana')
         return  (layer) ? layer.getEMSHotLink() : null;
       }
 
-
-      async _getUrlTemplateForEMSTMSLayer(tmsServiceConfig) {
+      async _getAttributesForEMSTMSLayer(isDesaturated, isDarkMode) {
         const tmsServices = await this._emsClient.getTMSServices();
+        const emsTileLayerId = mapConfig.emsTileLayerId;
+        let serviceId;
+        if (isDarkMode) {
+          serviceId = emsTileLayerId.dark;
+        } else {
+          if (isDesaturated) {
+            serviceId = emsTileLayerId.desaturated;
+          } else {
+            serviceId = emsTileLayerId.bright;
+          }
+        }
         const tmsService = tmsServices.find(service => {
-          return service.getId() === tmsServiceConfig.id;
+          return service.getId() === serviceId;
         });
-        return tmsService.getUrlTemplate();
+        return {
+          url: await tmsService.getUrlTemplate(),
+          minzoom: await tmsService.getMinZoom(),
+          maxzoom: await tmsService.getMaxZoom(),
+          attribution: await tmsService.getHTMLAttribution(),
+        };
       }
 
-      async getUrlTemplateForTMSLayer(tmsServiceConfig) {
-
+      async getAttributesForTMSLayer(tmsServiceConfig, isDesaturated, isDarkMode) {
```

Comment:
I think the response object here should include the `origin` attribute as well. it is used later on to determine whether or not to apply the CSS filtering